### ISSUE TITLE
Memory leak 

### ISSUE THREAD

**Issue by yacine-ser** [jmfayard/refreshVersions#536]
- [x] I have read the [guidelines for submitting issues](https://jmfayard.github.io/refreshVersions/contributing/submitting-issues/)
## 🐛 Describe the bug
When using the version 0.40.1, our builds fail with the following error
```
an FAILURE: Build failed with an exception.
* What went wrong:
GC overhead limit exceeded
```

After adding `+HeapDumpOnOutOfMemoryError` to the jvmargs and analysing the hprof we got the following report from [Eclipse MAT](https://www.eclipse.org/mat/) 

<img width="611" alt="Screenshot 2022-04-13 at 16 44 10" src="https://user-images.githubusercontent.com/76563526/163344577-f3b62828-da40-43ca-a03f-ba04f5280202.png">

Note : 
Current jvmargs in our gradle.properties
`org.gradle.jvmargs=-Xmx4096m -XX:+HeapDumpOnOutOfMemoryError`

## ⚠️ Current behavior

After multiple builds we have to restart the gradle deamon because it is taking most of the allowed allocated memory.
Not stopping it result in very slow builds and the above mentioned error.

## ✅ Expected behavior

That our build time remains consistent

## 💣 Steps to reproduce

N/A

## 📱 Tech info
N/A

**Comment by LouisCAD** [comment]
Hello, thank you for the report, I was suspecting that we had a memory leak issue but had failed to find where to look exactly.

I'll try to pinpoint where it's rooted, starting from the information you provided, and I hope we can find a solution for the next release.

**Comment by soywiz-invideo** [comment]
We are using refreshVersions 0.40.0, and we have a pretty big leak just running `./gradlew`, diffing two gradle heaps we found this (in the case it is helpful):

<img width="1766" alt="Screenshot 2022-04-21 at 18 25 34" src="https://user-images.githubusercontent.com/99644907/164511240-396180ed-fff9-4f7a-9be1-b9a05fb00776.png">

**Comment by LouisCAD** [comment]
I now have a reproducer in CI (see #546) and I'll be working on fixing this this week. Thank you a lot for the help in pinpointing this. It's related to code inside `UsedPluginsHolder` and triggers when refreshVersions manages the version of another Gradle plugin.

**Comment by LouisCAD** [comment]
Fixed in 0.40.2.
Thank you!
